lib/modules/ens/index.js

```diff
@@ -361,6 +362,37 @@ class ENS {
     this.embark.addProviderInit('names', code, shouldInit);
     this.embark.addConsoleProviderInit('names', code, shouldInit);
   }
+
+  ensResolve(name, cb) {
+    const self = this;
+    if(!self.configured) {
+      return cb(null, self.configured);
```

Comment:
I think it would be better to return an error if it is not configured. So the user knows that he tried to use ENS features, but he didn't configure it.

Reply:
what happens when an argument to a solidity function is a string and the user inputs a string that ends with .eth and also the ens module is not enabled ?

I think it would be better if we read the ens config in here. If its not enabled, then we return the callback with no error. If its not configured, then we return the callback with error as you have mentioned.
Let me know what you think ?

Reply:
I think in both the case where ENS is not enabled and not configured, we should return an error, but a clear error to tell if it's because of no configuration yet or not enabled

Reply:
I made the requested changes. Please review